core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -279,7 +280,7 @@ void onePhaseCommitRecords(Snapshot snapshot) throws CommitException {
           e,
           snapshot.getId());
     } catch (ExecutionException e) {
-      throw new CommitException(
+      throw new UnknownTransactionStatusException(
```

Comment:
`UnknownTransactionStatusException` should be thrown when an `ExecutionException` is caught, because it is unknown whether the mutations were applied or not in this case.